### PR TITLE
GF-28 - Search component and implement it on explore projects page

### PR DESCRIPTION
**Pull request by iamdavidpetric**

### CODE REVIEW

File: src/show/components/TextField/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const TextField = ({ value, onChange, placeholder }) => {
```
Comment (iamalextodoran) [inline src/show/components/TextField/index.js]: abc

File: src/show/pages/public/Explore/index.js [hunk 2]
```diff
@@ -3,19 +3,20 @@ import { useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 
 import categories from '../../../../process/constants';
-import { Button, ProgressBar } from '../../../components';
+import { Button, ProgressBar, TextField } from '../../../components';
 import { PROJECT_PATH } from '../../../../process/routes/paths';
```
Comment (iamalextodoran) [inline src/show/pages/public/Explore/index.js]: Length.

File: src/show/pages/public/Explore/index.js [hunk 3]
```diff
@@ -3,19 +3,20 @@ import { useEffect, useState } from 'react';
...
 
 const Explore = () => {
   const [projects, setProjects] = useState([]);
   const [selectedCategory, setSelectedCategory] = useState('');
+  const [search, setSearch] = useState('');
```
Comment (iamalextodoran) [inline src/show/pages/public/Explore/index.js]: Sort them by length.

File: src/show/pages/public/Explore/index.js [hunk 4]
```diff
@@ -24,13 +25,24 @@ const Explore = () => {
...
-      .get(`http://localhost:3000/v1/projects?category=${selectedCategory}`)
+      .get(
+        `http://localhost:3000/v1/projects?category=${selectedCategory}&search=${search}`
+      )
       .then(res => setProjects(res.data))
       .catch(err => err);
-  }, [selectedCategory]);
+  }, [selectedCategory, search]);
+
+  // className='flex justify-end items-center border border-white hover:border-primary-100 hover:px-3 w-64 h-8 rounded-full mr-5'
```
Comment (iamalextodoran) [inline src/show/pages/public/Explore/index.js]: No commented code like this should get into the main code. Delete.

File: src/show/components/TextField/index.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  return (
+    <div>
+      <input
+        value={value}
+        onChange={onChange}
+        placeholder={placeholder}
+        className='placeholder-gray-500 flex text-end justify-end transition-all duration-300 border border-white hover:border-primary-100 outline-none hover:px-3 w-64 h-8 rounded-full mr-5'
```
Comment (iamalextodoran) [inline src/show/components/TextField/index.js]: This should be a general implementation for a text field component yet you have the placeholder and value of inputs at the right. Do you want all your text field to look like this?

File: src/show/components/Button/index.js [hunk 6]
```diff
@@ -14,10 +14,10 @@ const Button = ({
     secondary: disabled ? `text-white` : `text-white hover:text-secondary-200`,
     outline: disabled
       ? `border-none rounded-full text-white`
-      : `border-2 rounded-full border-primary-50 text-primary-50 hover:border-white hover:bg-primary-50 hover:text-white`,
+      : `border rounded-full border-primary-50 text-primary-50 hover:border-white hover:bg-primary-50 hover:text-white`,
     'invert-outline': disabled
-      ? `border-2 rounded-full border-green-300 text-greeen-300 border-white bg-green-300 py-1 text-white`
-      : `border-2 rounded-full border-green-500 text-greeen-500 border-white bg-green-500 py-1 text-white hover:bg-green-600`
+      ? `border rounded-full border-green-300 text-greeen-300 border-white bg-green-300 py-1 text-white`
+      : `border rounded-full border-green-500 text-greeen-500 border-white bg-green-500 py-1 text-white hover:bg-green-600`
```
Comment (iamalextodoran) [inline src/show/components/Button/index.js]: You have back ticks yet I don't see the need for them here.

File: src/show/pages/public/Explore/index.js [hunk 7]
```diff
@@ -24,13 +25,24 @@ const Explore = () => {
...
 
   return (
-    <div className='h-full w-full mt-10'>
+    <div className='h-full w-full mt-2'>
+      <div className='flex items-end justify-end'>
+        <TextField
+          placeholder='Search'
+          value={search}
+          onChange={event => setSearch(event.target.value)}
```
Comment (iamalextodoran) [inline src/show/pages/public/Explore/index.js]: Use `e` instead of `event`, is shorter.

**Review by iamalextodoran — CHANGES_REQUESTED** [review]

File: src/show/components/TextField/index.js (hunk 1 above)
Comment (iamdavidpetric) [reply src/show/components/TextField/index.js]: allright

File: src/show/pages/public/Explore/index.js (hunk 2 above)
Comment (iamdavidpetric) [reply src/show/pages/public/Explore/index.js]: ok boi

File: src/show/pages/public/Explore/index.js (hunk 3 above)
Comment (iamdavidpetric) [reply src/show/pages/public/Explore/index.js]: Sa traiti

File: src/show/pages/public/Explore/index.js (hunk 4 above)
Comment (iamdavidpetric) [reply src/show/pages/public/Explore/index.js]: Ooops

File: src/show/components/Button/index.js (hunk 6 above)
Comment (iamdavidpetric) [reply src/show/components/Button/index.js]: done

File: src/show/pages/public/Explore/index.js (hunk 7 above)
Comment (iamdavidpetric) [reply src/show/pages/public/Explore/index.js]: doneeeeeeeeee

File: src/show/components/TextField/index.js (hunk 5 above)
Comment (iamdavidpetric) [reply src/show/components/TextField/index.js]: for you, everything in this world ;)

**Review by MIKALAKE — APPROVED** [review]

File: src/show/components/TextField/index.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+const TextField = ({ className, onChange, placeholder, value }) => {
+  return (
+    <div>
+      <input
+        value={value}
+        onChange={onChange}
+        placeholder={placeholder}
+        className={className}
```
Comment (iamalextodoran) [inline src/show/components/TextField/index.js]: Make a default text field style and if needed you can override it by the `className` prop.

**Review by MIKALAKE — APPROVED** [review]

**Pull request merged**